Update language around client version compatibility (#6939)
opensearch-project/documentation-website: _clients/index.md

  * [OpenSearch Hadoop client](https://github.com/opensearch-project/opensearch-hadoop) 

## Legacy clients

> Most clients that work with ElasticSearch OSS 7.10.2 *should* work with OpenSearch 1.x, but the latest versions of those clients might include license or version checks that artificially break compatibility. The table that follows includes recommendations for which client versions to use for best compatibility with OpenSearch 1.x. From OpenSearch 2.0 and later, no ElasticSearch clients are fully compatible with OpenSearch.

Reviewer: ```suggestion
Most clients that work with Elasticsearch OSS 7.10.2 *should* work with OpenSearch 1.x, but the latest versions of those clients might include license or version checks that artificially break compatibility. The table that follows includes recommendations for which client versions to use for best compatibility with OpenSearch 1.x. From OpenSearch 2.0 and later, no Elasticsearch clients are fully compatible with OpenSearch.
```

> Most clients that work with Elasticsearch OSS 7.10.2 *should* work with OpenSearch 1.x, but the latest versions of those clients might include license or version checks that artificially break compatibility. The table that follows includes recommendations for which client versions to use for best compatibility with OpenSearch 1.x. From OpenSearch 2.0 and later, no Elasticsearch clients are fully compatible with OpenSearch.

Author: ```suggestion
Clients that work with Elasticsearch OSS 7.10.2 should work with OpenSearch 1.x. The latest versions of those clients, however, might include license or version checks that artificially break compatibility. The following table provides recommendations for which client versions to use for best compatibility with OpenSearch 1.x. For OpenSearch 2.0 and later, no Elasticsearch clients are fully compatible with OpenSearch.
```

> While OpenSearch and ElasticSearch share a majority of core features, mixing and matching the client and server has a very high risk of errors and unexpected results. The issues will continue to get worse over time as OpenSearch and ElasticSearch diverge even further from each other. So, even if your ElasticSearch client still works with your OpenSearch cluster, we highly recommend using OpenSearch clients for OpenSearch clusters.

Reviewer: ```suggestion
While OpenSearch and Elasticsearch share a majority of core features, mixing and matching the client and server has a very high risk of errors and unexpected results. The issues will continue to get worse over time as OpenSearch and Elasticsearch diverge even further from each other. So, even if your Elasticsearch client still works with your OpenSearch cluster, we highly recommend using OpenSearch clients for OpenSearch clusters.
```

> While OpenSearch and ElasticSearch share a majority of core features, mixing and matching the client and server has a very high risk of errors and unexpected results. The issues will continue to get worse over time as OpenSearch and ElasticSearch diverge even further from each other. So, even if your ElasticSearch client still works with your OpenSearch cluster, we highly recommend using OpenSearch clients for OpenSearch clusters.

Reviewer 2: ```suggestion
While OpenSearch and Elasticsearch share several core features, mixing and matching the client and server has a high risk of errors and unexpected results. As OpenSearch and Elasticsearch continue to diverge, such risks may increase. Although your Elasticsearch client may continue working with your OpenSearch cluster, using OpenSearch clients for OpenSearch clusters is recommended.
```
